airflow/jobs/backfill_job.py

```diff
@@ -257,31 +258,30 @@ def _manage_executor_state(self, running):
                     ti.handle_failure(msg)
 
     @provide_session
-    def _get_dag_run(self, run_date, session=None):
+    def _get_dag_run(self, run_date: datetime, dag: DAG, session: Session = None):
         """
         Returns a dag run for the given run date, which will be matched to an existing
         dag run if available or create a new dag run otherwise. If the max_active_runs
         limit is reached, this function will return None.
 
         :param run_date: the execution date for the dag run
-        :type run_date: datetime.datetime
+        :param dag: DAG
         :param session: the database session object
-        :type session: sqlalchemy.orm.session.Session
```

Comment:
FYI: Docs don't matter so much on prviate (`_foo` functions) as I don't think they are rendered, but right now our docs rendering don't pick up the typing from type annotations :(

Reply:
Ahh that is too bad.

Reply:
Still, my IDEA loves the annotations 👍